2019/w16_Go_is_on_a_Trajectory_to_Become_the_Next_Enterprise_Programming_Language.md

```diff
@@ -0,0 +1,232 @@
+
+### Based on real-world Experience
+
+Go was created by experienced software industry veterans who have felt the pain from shortcomings of the existing languages for a long time. Rob Pike and Ken Thompson played a major role in the invention of Unix, C, and parts of Unicode many decades ago. Robert Griesemer has decades of experience around compilers and garbage collection after working on the V8 and HotSpot virtual machines for JavaScript and Java. Motivated by one too many times [they had to wait for their Google-sized C++/Java code bases to compile](http://radar.oreilly.com/2012/09/golang.html), they set out to create one more programming language that contains everything they learned through half a century of writing code.
+
+### Focus on large-scale Engineering
+
+Small engineering projects can be built successfully in pretty much any programming language. The really painful problems happen when thousands of developers collaborate under constant time pressure over decades on massive code bases containing tens of millions of lines of code. This leads to problems like:
+
+- long compile times interrupt development
+- the code base has been owned by several people/teams/departments/companies, mixing different programming styles
+- the company employs thousands of engineers, architects, testers, Ops specialists, auditors, interns, etc who need to understand the code base but have a wide range of coding experience
+- dependency on lots of external libraries or runtimes, some of them no longer existing in their original form
+- each line of code has been [rewritten an average of 10 times](https://www.ybrikman.com/writing/2018/08/12/the-10-to-1-rule-of-writing-and-programming) over the course of the lifetime of the code base, leaving scars, warts, and technical drift
+- incomplete documentation
+
+Go focuses on [alleviating these large-scale engineering pains](https://talks.golang.org/2012/splash.article), sometimes at the cost of making engineering in the small a bit more cumbersome, for example by requiring a few extra lines of code here and there.
+
+### Focus on Maintainability
+
+Go emphasizes offloading as much work as possible to automated code maintenance tools. The Go toolchain provides the most frequently needed functionality like formatting code and imports, finding definitions and usages of symbols, simple refactorings, and identification of code smells. Thanks to standardized code formatting and a single idiomatic way of doing things, machine-generated code changes look very close to human-generated changes in Go and use similar patterns, allowing more seamless collaboration of humans and machines.
+
+### Keep it Simple and Straightforward
+
+> _Junior programmers create simple solutions to simple problems. Senior programmers create complex solutions to complex problems. Great programmers find simple solutions to complex problems. — _[_Charles Connell_](http://www.chc-3.com/pub/beautifulsoftware_v10.htm)
```

Comment:
_Junior programmers create simple solutions to simple problems. Senior programmers create complex solutions to complex problems. Great programmers find simple solutions to complex problems. — [Charles Connell](http://www.chc-3.com/pub/beautifulsoftware_v10.htm)_